build.mill

```diff
@@ -795,7 +795,15 @@ object dist extends MillPublishJavaModule {
     val isWin = scala.util.Properties.isWin
     val outputPath = T.dest / (if (isWin) "run.bat" else "run")
 
-    os.write(outputPath, prependShellScript())
+    val prependShellScript0 = prependShellScript(
+      T.dest / "prependShellScript",
+      forkArgs(),
+      runClasspath().map(_.path),
+      pathingJar().path,
+      outputPath
+    )
+
+    os.write(outputPath, prependShellScript0)
```

Comment:
@alexarchambault what are these changes about? They don't seem related to BSP integration tests and their purpose doesn't seem obvious at a glance

Reply:
That's needed for [that step](https://github.com/com-lihaoyi/mill/blob/8b063c48d76837870af188ad48500bb739668f2b/bsp/src/mill/bsp/BSP.scala#L79-L83) of `mill.bsp.BSP/install` to work. This ensures that `mill.main.cli` has a valid value (path to the launcher), so that spawning a server from `.bsp/mill-bsp.json` works. Without this, `.bsp/mill-bsp.json` has `java.class.path` as executable, which doesn't work in the test (as that value is a `:`-separated list of JARs and directories, and not the launcher - the "pathing JARs" isn't there, only its list of JARs / directories)

Reply:
If we want to change where `mill.main.cli` points to, could we override it by passing in the `$MILL_MAIN_CLI` CLI variable? That would be something we can do directly in the integration test without needing to patch the launcher script

Reply:
That would let us keep `prependShellScript` as a `Task`, which would reduce the diff and keep things more easily inspectable

Reply:
Sure, the latest push does that